src/main/java/uk/gov/hmcts/reform/notificationservice/service/NotificationMessageProcessor.java

```diff
@@ -0,0 +1,158 @@
+package uk.gov.hmcts.reform.notificationservice.service;
+
+import com.microsoft.azure.servicebus.IMessage;
+import com.microsoft.azure.servicebus.IMessageReceiver;
+import com.microsoft.azure.servicebus.primitives.ServiceBusException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.notificationservice.exception.InvalidMessageException;
+import uk.gov.hmcts.reform.notificationservice.exception.UnknownMessageProcessingResultException;
+
+@Service
+public class NotificationMessageProcessor {
+
+    private static final Logger log = LoggerFactory.getLogger(NotificationMessageProcessor.class);
+
+    private final IMessageReceiver messageReceiver;
+    private final NotificationMessageHandler notificationMessageHandler;
+    private final NotificationMessageParser notificationMessageParser;
+    private final int maxDeliveryCount;
+
+    public NotificationMessageProcessor(
+        IMessageReceiver messageReceiver,
+        NotificationMessageHandler notificationMessageHandler,
+        NotificationMessageParser notificationMessageParser,
+        @Value("${queue.notifications.max-delivery-count}") int maxDeliveryCount
+    ) {
+        this.messageReceiver = messageReceiver;
+        this.notificationMessageHandler = notificationMessageHandler;
+        this.notificationMessageParser = notificationMessageParser;
+        this.maxDeliveryCount = maxDeliveryCount;
+    }
+
+
+    private enum MessageProcessingResult {
+        SUCCESS,
+        UNRECOVERABLE_FAILURE,
+        POTENTIALLY_RECOVERABLE_FAILURE
+    }
```

Comment:
Typically, inner classes, etc., are placed at the bottom of the file

Reply:
moved